### ISSUE TITLE
Make Department Format Reports

### ISSUE THREAD

**Issue by amhanson9** [uga-libraries/format-report#9]
Having a summary of everything in ARCHive, which the script currently does, gives us documentation of our big picture but is not enough nuance for departments to act on.

Make a department report with a summary of all of their formats, focusing on risk. Work with Emmeline on what is most useful for a department doing risk mediation. The focus is on a spreadsheet with data summaries, but also include a brief narrative report to provide an overview. The narrative is made by hand using data from the spreadsheet generated by the script.

**Comment by amhanson9** [comment]
Should the department report include a narrative and/or visualizations? Or is a spreadsheet with the risk data and summaries enough?

**Comment by amhanson9** [comment]
From 8/29 meeting with Emmeline: 

Primary spreadsheet is organized by AIP and has Collection ID, AIP ID, Format Name, Format Version, PRONOM URL (combine registry name and key), NARA risk level, and NARA plan. Need to organize by AIP to have the collection information, which is important for her prioritization. It means there will not be size or file count information.

Summary by collection with the percentage of files at each NARA risk level (low, moderate, high, no matched) to identify collections with more risk.

If possible, by collection and then by AIP with columns that are file count for each format (combine name and version). The pivot table she made color coded the formats for risk level.

If time, do a proof of concept for identify formats with risk levels that increased from the previous analysis. Recent changes indicate a potential urgency to review the format, such as a limited window the vendor may make a migration tool available.

A brief narrative summary could be helpful, with visualizations to show current risk, change in risk, maybe the most common.

**Comment by amhanson9** [comment]
Resources used:
https://stackoverflow.com/questions/62330833/pandas-split-dataframe-by-unique-column-value
https://stackoverflow.com/questions/16729574/how-can-i-get-a-value-from-a-cell-of-a-dataframe
https://stackoverflow.com/questions/45487410/how-to-calculate-counts-on-pandas-pivot-table
https://stackoverflow.com/questions/23667369/drop-all-duplicate-rows-across-multiple-columns-in-python-pandas
https://stackoverflow.com/questions/40427943/how-do-i-change-a-single-index-value-in-pandas-dataframe
https://stackoverflow.com/questions/54334674/how-to-use-pandas-to-pivot-a-table-while-changing-values-to-boolean
https://stackoverflow.com/questions/53934834/how-to-sort-the-values-in-pandas-pivot-table
https://stackoverflow.com/questions/41476150/remove-or-replace-spaces-in-column-names
https://www.statology.org/pandas-create-new-column-based-on-multiple-condition-if-else/

**Comment by amhanson9** [comment]
We decided to use format name and version when determining unique format identifications. We did not include the PUID, since that is often from files being identified as the format with a PUID by one tool and without the PUID by another tool, inflating the number of files. Version is important, since different versions of the same format can have a different risk. We'll revisit the decision if we find it is common for the same file to have multiple possible versions and therefore it is resulting in inflated counts.
